src/chi/javascript/components/mobile-navigation.js

```diff
@@ -0,0 +1,458 @@
+
+    this._initFirstLevelDrawer();
+    this._addEventHandler(
+      this._drawersContainer,
+      'click',
+      (e) => {
+        e.stopPropagation();
+        this._handleClickOnSecondLevelDrawer(e);
+      });
+  }
+
+  static get componentType() {
+    return COMPONENT_TYPE;
+  }
+
+  _initFirstLevelDrawer() {
+    this._mobileNavElement = this._locateMobileNav();
+    const firstLevelDrawer = this._mobileNavElement.querySelector(`.${FIRST_LEVEL_DRAWER_CLASS}`);
+    const firstLevelMenuItems = this._mobileNavElement.querySelectorAll(`.${DRAWER_CLASS}.${FIRST_LEVEL_DRAWER_CLASS} .${FIRST_LEVEL_DRAWER_CONTENT_CLASS} .${DRAWER_CONTENT_CLASS} .${CHI_TABS_CLASS} li a`);
+    const firstLevelDropdownTriggers = this._mobileNavElement.querySelectorAll(`.${DRAWER_CLASS}.${FIRST_LEVEL_DRAWER_CLASS} .${DROPDOWN_TRIGGER}`);
+    const self = this;
+
+    if (firstLevelDrawer) {
+      const firstLevelCloseButton = firstLevelDrawer.querySelector(`.${chi.classes.CLOSE}`);
+
+      this._firstLevelDrawer = Drawer.factory(this._elem, {
+        animated: true,
+        target: firstLevelDrawer
+      });
+      this._addEventHandler(
+        firstLevelCloseButton,
+        'click',
+        () => {
+          self._hideMobileNav();
+        }
+      );
+    }
+
+    Array.prototype.forEach.call(
+      firstLevelMenuItems,
+      (menuItem) => {
+        const drawerId = menuItem.getAttribute('href');
+        const drawer = document.querySelector(`.chi-drawer${drawerId}`);
+
+        if (drawer) {
+          const returnBackButton = drawer.querySelector(`.${DRAWER_HEADER_CLASS} button.${chi.classes.BACK}`);
+          const firstLevelContent = self._mobileNavElement.querySelector(`.${FIRST_LEVEL_DRAWER_CONTENT_CLASS}`);
+
+          this._addEventHandler(
+            menuItem,
+            'click',
+            function(e) {
+              e.preventDefault();
+              Util.addClass(drawer, chi.classes.ACTIVE);
+              Util.addClass(firstLevelContent, SECOND_LEVEL_SHOWN_CLASS);
+            });
+
+          this._addEventHandler(
+            returnBackButton,
+            'click',
+            function(e) {
+              e.stopPropagation();
+              Util.removeClass(drawer, chi.classes.ACTIVE);
+              self.resetActiveDrawerMenuItem();
+              Util.removeClass(firstLevelContent, SECOND_LEVEL_SHOWN_CLASS);
+            });
+          Util.addClass(menuItem, DRAWER_SUBITEM_LIST_TRIGGER_CLASS);
+          this._drawers.push(drawer);
+        } else {
+          this._addEventHandler(
+            menuItem,
+            'click',
+            () => self._activateFirstLevelMenuItem(menuItem)
+          );
+        }
+      }
+    );
+    Array.prototype.forEach.call(
+      firstLevelDropdownTriggers,
+      (dropdown) => {
+        const dropdownElemId = dropdown.getAttribute('data-target');
+
+        Dropdown.factory(dropdown, { dropdownElem: document.querySelector(`#${dropdownElemId}`) });
```

Comment:
You need to preserve the reference to the instantiated dropdown so you can dispose it once you don't need it any longer.